Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -646,558 +586,373 @@ protected virtual void Dispose(bool disposing)
-        /// </summary>
-        /// <param name="address">Address to convert.</param>
-        /// <returns>URI safe conversion of the address.</returns>
-        private static string FormatIP6String(IPAddress address)
+        /// <inheritdoc/>
+        public bool HasRemoteAccess(IPAddress remoteIP)
         {
-            var str = address.ToString();
-            if (address.AddressFamily == AddressFamily.InterNetworkV6)
+            var config = _configurationManager.GetNetworkConfiguration();
+            if (config.EnableRemoteAccess)
             {
-                int i = str.IndexOf("%", StringComparison.OrdinalIgnoreCase);
-                if (i != -1)
+                // Comma separated list of IP addresses or IP/netmask entries for networks that will be allowed to connect remotely.
+                // If left blank, all remote addresses will be allowed.
+                if (_remoteAddressFilter.Any() && !_lanSubnets.Any(x => x.Contains(remoteIP)))
                 {
-                    str = str.Substring(0, i);
-                }
+                    // remoteAddressFilter is a whitelist or blacklist.
+                    var matches = _remoteAddressFilter.Count(remoteNetwork => remoteNetwork.Contains(remoteIP));
+                    if ((!config.IsRemoteIPFilterBlacklist && matches > 0)
+                        || (config.IsRemoteIPFilterBlacklist && matches == 0))
+                    {
+                        return true;
+                    }
 
-                return $"[{str}]";
+                    return false;
+                }
+            }
+            else if (!_lanSubnets.Any(x => x.Contains(remoteIP)))
+            {
+                // Remote not enabled. So everyone should be LAN.
+                return false;
             }
 
-            return str;
+            return true;
         }
 
-        private void ConfigurationUpdated(object? sender, ConfigurationUpdateEventArgs evt)
+        /// <inheritdoc/>
+        public IReadOnlyList<PhysicalAddress> GetMacAddresses()
         {
-            if (evt.Key.Equals(NetworkConfigurationStore.StoreKey, StringComparison.Ordinal))
-            {
-                UpdateSettings((NetworkConfiguration)evt.NewConfiguration);
-            }
+            // Populated in construction - so always has values.
+            return _macAddresses;
         }
 
-        /// <summary>
-        /// Checks the string to see if it matches any interface names.
-        /// </summary>
-        /// <param name="token">String to check.</param>
-        /// <param name="index">Interface index numbers that match.</param>
-        /// <returns><c>true</c> if an interface name matches the token, <c>False</c> otherwise.</returns>
-        private bool TryGetInterfaces(string token, [NotNullWhen(true)] out List<int>? index)
+        /// <inheritdoc/>
+        public IReadOnlyList<IPData> GetLoopbacks()
         {
-            index = null;
+            if (!(IsIPv4Enabled && IsIPv4Enabled))
```

Comment:
Is the other one supposed to be IPv6? If so, I prefer having it as `if (!IsIPv4Enabled || !IsIPv6Enabled)`. I am not a fan of negating a whole expression

Reply:
It should return an empty array if both are disabled.